### PR TITLE
Interactive window- don't leave space for insert toolbar

### PR DESCRIPTION
**Pull request by r3m0t**
Interactive window is more compact than Notebook now, by not leaving space for the insert cell toolbar(which has no items anyway in the interactive window)

This is cleaned up from #164760

### CODE REVIEW

**Review by amunger — DISMISSED** [review]

File: src/vs/workbench/contrib/notebook/browser/notebookOptions.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ export class NotebookOptions extends Disposable {
 		const cellToolbarInteraction = overrides?.cellToolbarInteraction ?? this.configurationService.getValue<string>(NotebookSetting.cellToolbarVisibility);
 		const compactView = this.configurationService.getValue<boolean | undefined>(NotebookSetting.compactView) ?? true;
 		const focusIndicator = this._computeFocusIndicatorOption();
-		const insertToolbarPosition = this._computeInsertToolbarPositionOption();
+		const insertToolbarBetweenCells = overrides?.insertToolbarBetweenCells ?? this._computeInsertToolbarBetweenCellsOption();
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/notebookOptions.ts:148]: should we check if the notebook document is readonly or not?

Comment (r3m0t) [reply src/vs/workbench/contrib/notebook/browser/notebookOptions.ts:148]: Yep, it is a bigger change now but it looks great. I used `Show Contents` on local history timeline/git timeline to see a readonly jupyter notebook.

File: src/vs/workbench/contrib/notebook/browser/notebookOptions.ts [hunk 2]
```diff
@@ -547,34 +548,37 @@ export class NotebookOptions extends Disposable {
...
 	}
 
-	private _computeBottomToolbarDimensions(compactView: boolean, insertToolbarPosition: 'betweenCells' | 'notebookToolbar' | 'both' | 'hidden', insertToolbarAlignment: 'left' | 'center', cellToolbar: 'right' | 'left' | 'hidden'): { bottomToolbarGap: number; bottomToolbarHeight: number } {
+	private _computeBottomToolbarDimensions(compactView: boolean, insertToolbarBetweenCells: boolean, insertToolbarAlignment: 'left' | 'center', cellToolbar: 'right' | 'left' | 'hidden'): { bottomToolbarGap: number; bottomToolbarHeight: number } {
+		if (!insertToolbarBetweenCells) {
+			return compactView ? {
+				bottomToolbarGap: 0,
+				bottomToolbarHeight: 0
+			} : {
+				bottomToolbarGap: 12,
```
Comment (r3m0t) [inline src/vs/workbench/contrib/notebook/browser/notebookOptions.ts:557]: this is enough gap for the `cellToolbar` to have a gap between the cell and the previous cell (in the case the previous cell has no output)

File: src/vs/workbench/contrib/notebook/browser/notebookOptions.ts (hunk 1 above)
Comment (amunger) [reply src/vs/workbench/contrib/notebook/browser/notebookOptions.ts:148]: `insertToolbarBetweenCells` should be set to false here if the notebook is readonly. That would eliminate the need to pass around the readonly parameter in all those other methods.

Comment (r3m0t) [reply src/vs/workbench/contrib/notebook/browser/notebookOptions.ts:148]: That information comes from the model, `NotebookEditorWidget` doesn't have the model in its constructor, only later when `NotebookEditorWidget.setModel` is called (which can be done multiple times)

Here's the call stack when using Show Contents on a git revision (`creationOptions` is ```{
  menuIds: {...
  },
  cellEditorContributions: [
...]
}```)

image

Then setModel-
image

Then setOptions with isReadOnly set-

image

**Review by amunger — APPROVED** [review]

**Review by Tyriar — APPROVED** [review]

**Comment by r3m0t** [comment]
Thanks @amunger, users at my workplace will be very happy. During the change, I noticed #182120 as well.

**Pull request merged**